packages/camera/camera_platform_interface/lib/src/method_channel/method_channel_camera.dart

```diff
@@ -2,11 +2,214 @@
+
+  Stream<CameraEvent> _events(int cameraId) =>
+      _cameraEventStreamController.stream
+          .where((event) => event.cameraId == cameraId);
+
+  @override
+  Future<List<CameraDescription>> availableCameras() async {
+    try {
+      final List<Map<dynamic, dynamic>> cameras = await _channel
+          .invokeListMethod<Map<dynamic, dynamic>>('availableCameras');
+      return cameras.map((Map<dynamic, dynamic> camera) {
+        return CameraDescription(
+          name: camera['name'],
+          lensDirection: parseCameraLensDirection(camera['lensFacing']),
+          sensorOrientation: camera['sensorOrientation'],
+        );
+      }).toList();
+    } on PlatformException catch (e) {
+      throw CameraException(e.code, e.message);
+    }
+  }
+
+  @override
+  Future<int> initializeCamera(
+    CameraDescription cameraDescription,
+    ResolutionPreset resolutionPreset, {
+    bool enableAudio,
+  }) async {
+    int _textureId;
+    try {
+      final Map<String, dynamic> reply =
+          await _channel.invokeMapMethod<String, dynamic>(
+        'initialize',
+        <String, dynamic>{
+          'cameraName': cameraDescription.name,
+          'resolutionPreset': resolutionPreset != null
+              ? _serializeResolutionPreset(resolutionPreset)
+              : null,
+          'enableAudio': enableAudio,
+        },
+      );
+      _textureId = reply['textureId'];
+    } on PlatformException catch (e) {
+      throw CameraException(e.code, e.message);
+    }
+    if (!_channels.containsKey(_textureId)) {
+      final channel =
+          MethodChannel('flutter.io/cameraPlugin/camera$_textureId');
+      channel.setMethodCallHandler(
+          (MethodCall call) => handleMethodCall(call, _textureId));
+      _channels[_textureId] = channel;
+    }
+    return _textureId;
+  }
+
+  @override
+  Future<void> dispose(int cameraId) async {
+    await _channel.invokeMethod<void>(
+      'dispose',
+      <String, dynamic>{'textureId': cameraId},
+    );
+    _channels.remove(cameraId);
+  }
+
+  @override
+  Stream<ResolutionChangedEvent> onResolutionChanged(int cameraId) {
+    assert(_channels.containsKey(cameraId));
+    return _events(cameraId).whereType<ResolutionChangedEvent>();
+  }
+
+  @override
+  Stream<CameraClosingEvent> onCameraClosing(int cameraId) {
+    assert(_channels.containsKey(cameraId));
+    return _events(cameraId).whereType<CameraClosingEvent>();
+  }
+
+  @override
+  Stream<CameraErrorEvent> onCameraError(int cameraId) {
+    assert(_channels.containsKey(cameraId));
+    return _events(cameraId).whereType<CameraErrorEvent>();
+  }
+
+  @override
+  Future<XFile> takePicture(int cameraId) async {
+    String path = await _channel.invokeMethod<String>(
+      'takePicture',
+      <String, dynamic>{'textureId': cameraId},
+    );
+    return XFile(path);
+  }
+
+  @override
+  Future<void> prepareForVideoRecording() async {
+    await _channel.invokeMethod<void>('prepareForVideoRecording');
+  }
```

Comment:
You don't have to await this here, you could simply return the future from `_channel.invokeMethod`:
```suggestion
  Future<void> prepareForVideoRecording() {
    return _channel.invokeMethod<void>('prepareForVideoRecording');
  }
```

or even shorter:
```suggestion
  Future<void> prepareForVideoRecording() =>
    _channel.invokeMethod<void>('prepareForVideoRecording');
```